src/ovp_pipeline/context_binder.py

```
@@ -0,0 +1,679 @@
+    * Anchor section (only when non-empty)
+    * Retrieval section (only when non-empty)
+
+    Empty sections are skipped so a standalone inquiry doesn't
+    waste tokens on placeholder headings.
+    """
+    parts: list[str] = []
+    if prefix:
+        parts.append(prefix.rstrip())
+    if anchor.included_anchor:
+        parts.append(
+            f"# Anchor — {anchor.kind}: {anchor.ref}\n\n" f"{anchor.included_anchor.rstrip()}"
```

Ruff (0.15.12)

[warning] Implicitly concatenated string literals on one line

Combine string literals

(ISC001)